src/fastapi_poe/base.py

```diff
@@ -629,6 +645,263 @@ def make_prompt_author_role_alternated(
 
         return new_messages
 
+    async def capture_cost(
+        self,
+        request: QueryRequest,
+        amounts: Union[List[CostItem], CostItem],
+        access_key: Optional[str] = None,
+        base_url: str = "https://api.poe.com/",
+    ) -> None:
+        """
+
+        Used to capture a cost for monetized creators.
+
+        #### Parameters:
+        - `request` (`QueryRequest`): The currently handlded QueryRequest object.
+        - `amounts` (`Union[list[CostItem], CostItem]`): The to be captured amounts.
+        - `access_key` (`str`): The access_key corresponding to your bot. This is needed to ensure
+        that capture requests are coming from an authorized source.
+
+        """
+
+        if self.access_key:
+            if access_key:
+                warnings.warn(
```

Comment:
probably can remove the access_key param and force developers to set an access_key on the bot level. I believe this param + deprecation warning only exists on other functions for backwards compatibility reasons, but seeing as this is a new function, its best not to have it at all.